### ISSUE TITLE
AWS Otel middleware not populating attributes for DynamoDB

### ISSUE THREAD

**Issue by noliva** [open-telemetry/opentelemetry-go-contrib#2260]
# Background

When using the Otel middleware, no attributes like tableName get populated in the span. 

This is how I'm instrumenting the code.

```go
awsCfg, err := awsConfig.LoadDefaultConfig(ctx)
if err != nil {
    panic("configuration error, " + err.Error())
}

otelaws.AppendMiddlewares(&awsCfg.APIOptions)

client := dynamodb.NewFromConfig(cfg)
```

When the middleware is trying to detect the `ServiceID` [here](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/aws.go#L62) the function returns `DynamoDB`, however, when it later looks for the `DefaultAttributeSetter` it can not get the `DynamoDBAttributeSetter` because the `servicemap` [looks like this](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/attributes.go#L34)

```go
var servicemap = map[string]AttributeSetter{
	"dynamodb": DynamoDBAttributeSetter,
}
```

The key of the map references the `signingNameKey` and not the `serviceIDKey`.

# Proposal

I believe that the servicemap should be updated to reflect the `serviceIDKey`, so it would look like this.

```go
var servicemap = map[string]AttributeSetter{
	"DynamoDB": DynamoDBAttributeSetter,
}
```

Is that the case or am I doing something wrong in the process?

**Comment by Aneurysm9** [comment]
I think your analysis is correct.  Can you open a PR to update this?  As the [`dynamodb.ServiceID`](https://github.com/aws/aws-sdk-go-v2/blob/69119818ef63fe807b5613a9f554f90308480409/service/dynamodb/api_client.go#L32) constant is exported it might be best to use that rather than a string literal.

**Comment by noliva** [comment]
Thank you for looking into this, yes, I will open a PR to address this.

**Comment by MrAlias** [comment]
This looks to have been resolved by #2272
